eth/vm/state.py

```diff
@@ -175,14 +200,16 @@ def get_ancestor_hash(self, block_number):
             ancestor_depth >= len(self.execution_context.prev_hashes)
         )
         if is_ancestor_depth_out_of_range:
-            return b''
+            return cast(Hash32, b'')
```

Comment:
Don't cast here, do a `Hash32(b'')` instead. In general: a cast always means that you are smarter than the type checker which means you can even cast something silly as `cast(Hash32, "foobar")`. So, if we are creating the type here, we don't need the cast and can simply use the constructor.